docs/pages/guides/adding-a-new-signer-to-identity-kit.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: "Adding a new signer to Identity Kit"
+date: "2024-08-05"
+authors:
+  - name: "Dmytro Sheremet"
+---
+
+# Adding a New Signer to Identity Kit
+
+To add your wallet as a signer to Identity Kit, follow these four steps:
+
+1. Clone the Identity Kit repository.
+2. Add your wallet configuration.
+3. Create a pull request.
+4. Wait for approval and then merge the pull request.
+
+## Steps
+
+### 1. Clone the Identity Kit repository
+
+First, clone the Identity Kit repository to your local machine. Open your terminal and run the
+following command:
+
+```bash
+git clone https://github.com/internet-identity-labs/identitykit.git
+```
+
+### 2. Add your wallet configuration
+
+Open the `packages/identitykit/src/lib/signers.ts` file in your preferred code editor and add your
+wallet configuration. Ensure that each field is correctly filled out:
+
+- id: A unique identifier for your wallet. Make sure it is unique to avoid conflicts.
+- providerUrl: The URL to connect to your wallet. This URL should be compatible with ICRC standards.
+- transportType: The method your wallet uses to communicate with the client's DApp. If your wallet
+  is an extension, use `TransportType.EXTENSION`. For web DApps, you can choose between
+  `TransportType.POPUP` and `TransportType.IFRAME`, each with its own advantages and disadvantages.
+- popupHeight and popupWidth: The dimensions of the sign-in window.
+- label: The display name of your wallet.
+- icon: A base64-encoded SVG image representing your wallet.
+
+Here is an example configuration:
+
+```typescript
+const YourWalletSigner: SignerConfig = {
+  id: "YourWalletSignerIdentifier",
+  providerUrl: "https://your-wallet-rpc-url.com",
+  transportType: TransportType.POPUP,
+  popupHeight: 640,
+  popupWidth: 450,
+  label: "Your Wallet Name",
+  icon: "data:image/svg+xml;base64,PHN2ZyB3aWR0ac+Cg==",
+}
+```
+
+### 3. Create a pull request
+
+After adding your wallet configuration, commit your changes and push them to the repository. Then,
+create a pull request (PR).
+
+### 4. Wait for approval and then merge the pull request
+
+Wait for the maintainers to review and approve your pull request. Once approved, you can merge it
+into the main branch.
```

Comment:
Once approved, it will be merged into the `main` branch, and the new version of the [@nfid/identitykit NPM package](https://www.npmjs.com/package/@nfid/identitykit) will be released.

Reply:
i'll prefer to keep merging to the `main` and releases in house (;

Reply:
Fixed, thanks!